scripts/testing-utils.js

```diff
@@ -115,8 +119,171 @@ function launchPackagerInSeparateWindow(folderPath) {
+    const options = {
+      method: 'GET',
+      url: 'https://circleci.com/api/v2/project/gh/facebook/react-native/pipeline',
+      qs: {
+        branch: branchName,
+      },
+      headers: this.circleCIHeaders,
+    };
+
+    const response = await request(options);
+    if (response.error) {
+      throw new Error(error);
+    }
+
+    const lastPipeline = JSON.parse(response.body).items[0];
+    return {id: lastPipeline.id, number: lastPipeline.number};
+  }
+
+  async #getSpecificWorkflow(pipelineId, workflowName) {
+    const options = {
+      method: 'GET',
+      url: `https://circleci.com/api/v2/pipeline/${pipelineId}/workflow`,
+      headers: this.circleCIHeaders,
+    };
+    const response = await request(options);
+    if (response.error) {
+      throw new Error(error);
+    }
+
+    const body = JSON.parse(response.body);
+    return body.items.find(workflow => workflow.name === workflowName);
+  }
+
+  async #getPackageAndReleaseWorkflow(pipelineId) {
+    return this.#getSpecificWorkflow(
+      pipelineId,
+      'package_and_publish_release_dryrun',
+    );
+  }
+
+  async #getTestsWorkflow(pipelineId) {
+    return this.#getSpecificWorkflow(pipelineId, 'tests');
+  }
+
+  async #getCircleCIJobs(workflowId) {
+    const options = {
+      method: 'GET',
+      url: `https://circleci.com/api/v2/workflow/${workflowId}/job`,
+      headers: this.circleCIHeaders,
+    };
+    const response = await request(options);
+    if (response.error) {
+      throw new Error(error);
+    }
+
+    const body = JSON.parse(response.body);
+    return body.items;
+  }
+
+  async #getJobsArtifacts(jobNumber) {
+    const options = {
+      method: 'GET',
+      url: `https://circleci.com/api/v2/project/gh/facebook/react-native/${jobNumber}/artifacts`,
+      headers: this.circleCIHeaders,
+    };
+    const response = await request(options);
+    if (response.error) {
+      throw new Error(error);
+    }
+
+    const body = JSON.parse(response.body);
+    return body.items;
+  }
+
+  async #findUrlForJob(jobName, artifactPath) {
+    const job = this.jobs.find(j => j.name === jobName);
+    const artifacts = await this.#getJobsArtifacts(job.job_number);
+    return artifacts.find(artifact => artifact.path.indexOf(artifactPath) > -1)
+      .url;
+  }
+
+  async artifactURLHermesDebug() {
+    return this.#findUrlForJob(
+      'build_hermes_macos-Debug',
+      'hermes-ios-debug.tar.gz',
+    );
+  }
+
+  async artifactURLForMavenLocal() {
+    return this.#findUrlForJob(
+      'build_and_publish_npm_package-2',
+      'maven-local.zip',
+    );
+  }
+
+  async artifactURLForPackagedReactNative() {
+    return this.#findUrlForJob(
+      'build_and_publish_npm_package-2',
+      'react-native-1000.0.0-',
+    );
+  }
+
+  async artifactURLForHermesRNTesterAPK() {
+    return this.#findUrlForJob(
+      'test_android',
+      'rntester-apk/hermes/release/app-hermes-arm64-v8a-release.apk',
```

Comment:
is the `-arm64-v8a-` version the one that will always work on both Intel and M1 machines? Or is it just for apple chips? we should have a comment here if it's bound to a specific setup.

Reply:
uhm... this is what works with the simulator. I think we can specify the simulator architecture, and I double checked with @cortinico  that this should be the right APK to use. 
If that's machine dependent, we can improve the script as soon as someone with a non M1 machine joins the crew... (probably with 0.73 and @Titozzz?)

Reply:
If you want to do it more sophisticated, you can run a `adb` command to ask the connected device what's the architecture and fetch the correct APK from device.